### PR TITLE
Replace fastfloat dependency with std::from_chars

### PR DESCRIPTION
**Pull request by mavam**
## Summary

This PR removes the external `fast_float` dependency and replaces it with `std::from_chars` from the C++ standard library for double parsing.

### Changes

- **Removed fastfloat submodule** from `libtenzir/aux/fast_float`
- **Implemented custom double parser** in `libtenzir/src/concept/parseable/numeric/real.cpp`:
  - Fast path for '.' separator: directly uses `std::from_chars` with no pre-scanning
  - Slow path for ',' separator: scans, copies to buffer with separator replacement, then parses
  - Handles special values (inf, nan, infinity) case-insensitively
  - Handles leading '+' sign (which `std::from_chars` rejects)
  - Treats `result_out_of_range` as success (overflow → infinity, underflow → zero)
- **Added comprehensive test coverage**:
  - Integration tests in `test/tests/functions/float/parse-edge-cases.tql`
  - Unit tests for comma separator, special values, leading plus sign, extreme exponents, and error cases
- **Updated build system** (CMake, Nix) to remove fastfloat references

### Motivation

- Reduces external dependencies
- Uses well-tested standard library implementation
- Maintains the same parsing accuracy and performance

## Test plan

- [x] Unit tests pass (95/95)
- [x] Integration tests pass (590/594, with 1 pre-existing failure and 3 skipped)

🤖 Generated with [Claude Code](https://claude.ai/code)

### CODE REVIEW

**Review by lava — COMMENTED** [review]
I didn't look at the fallback for now; the fast_float path looks reasonable overall.

**Comment by mavam** [comment]
@tobim Could you clarify which `nixpkgs` bump you had in mind?

The current PR CI is still using the pin from `.github/workflows/nix-build-job.yaml`:
`0d70460758949966e91d9ecb823b821f963cefbb`

And the failing Darwin Nix job is still building against `libcxx-19.1.2+apple-sdk-15.5`, so floating-point `std::from_chars` is not available there yet.

If there is another PR/commit/branch that updates the CI pin or Darwin toolchain, I can rebase/cherry-pick that and retry.

File: cmake/TenzirPackage.cmake [hunk 1]
```diff
@@ -134,9 +134,6 @@ cpack_add_component(
   DISPLAY_NAME "Runtime"
   DESCRIPTION "Runtime files for Tenzir"
   REQUIRED)
-# TODO: Remove all Unspecified install components. The vendored fast_float
-# library always installs itself; once fastfloat/fast_float#142 is merged we can
-# update the vendored library and reconsider this.
```
Comment (tobim) [inline cmake/TenzirPackage.cmake:139]: There should be a replacement comment here instead. At the time of writing the next `cpack_add_component` call was added for fast_float in particular, but we have a couple of other vendoered and bundled dependencies, and we don't know whether some of them do not use explicit installation components.
This PR should only update the comment tho, don't expand the scope of the change any more.

**Review by tobim — APPROVED** [review]
